app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt

```diff
@@ -3,36 +3,174 @@ package org.dhis2.usescases.searchte
 import android.content.Intent
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.rule.ActivityTestRule
+import org.dhis2.R
 import org.dhis2.usescases.BaseTest
+import org.dhis2.usescases.flow.teiFlow.entity.DateRegistrationUIModel
 import org.dhis2.usescases.searchTrackEntity.SearchTEActivity
+import org.dhis2.usescases.searchte.entity.DisplayListFieldsUIModel
+import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class SearchTETest : BaseTest() {
 
-    private lateinit var searchTETest: SearchTETest
-
     @get:Rule
     val rule = ActivityTestRule(SearchTEActivity::class.java, false, false)
 
     @Test
-    fun openSearchTE() {
+    fun shouldSuccessfullySearchByName() {
+        val firstName = "Tim"
+        val firstNamePosition = 0
+        val filterCount = "1"
+
+        prepareChildProgrammeIntentAndLaunchActivity()
+
+        searchTeiRobot {
+            searchByPosition(firstName, firstNamePosition)
```

Comment:
Do you think _**typeAttribute(firstName, firstNamePosition)**_ or _**typeAttributeAtPosition(firstName, firstNamePosition)**_ are more accurate than "searchByPosition"

Reply:
great I'll rename the function

Reply:
thanks @Sharmyn28